integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/noarkexchange/receive/EDUCoreConverter.java

```diff
@@ -56,14 +59,16 @@ public EDUCore unmarshallFrom(byte[] message) {
             PayloadConverter payloadConverter;
             if (eduCore.getMessageType() == EDUCore.MessageType.EDU) {
                 payloadConverter = new PayloadConverter<>(MeldingType.class);
-                eduCore.setPayload(payloadConverter.unmarshallFrom(((String)eduCore.getPayload()).getBytes()));
+                eduCore.setPayload(payloadConverter.unmarshallFrom(((String)eduCore.getPayload()).getBytes("UTF-8")));
             } else {
                 payloadConverter = new PayloadConverter<>(AppReceiptType.class);
-                eduCore.setPayload(payloadConverter.unmarshallFrom(((String)eduCore.getPayload()).getBytes()));
+                eduCore.setPayload(payloadConverter.unmarshallFrom(((String)eduCore.getPayload()).getBytes("UTF-8")));
             }
             return eduCore;
         } catch (JAXBException e) {
             throw new RuntimeException("Unable to create unmarshaller for " + EDUCore.class, e);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS00112)